modules/programs/nnn.nix

```
@@ -0,0 +1,132 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.nnn;
+
+  renderSetting = key: value: "${key}:${value}";
+
+  renderSettings = settings:
+    concatStringsSep ";" (mapAttrsToList renderSetting settings);
+
+  pluginModule = types.submodule ({ config, ... }: {
+    options = {
+      src = mkOption {
+        type = with types; nullOr path;
+        example = literalExample ''
+          (pkgs.fetchFromGitHub {
+            owner = "jarun";
+            repo = "nnn";
+            rev = "v4.0";
+            sha256 = "sha256-Hpc8YaJeAzJoEi7aJ6DntH2VLkoR6ToP6tPYn3llR7k=";
+          }) + "/plugins";
+        '';
+        default = null;
+        description = ''
+          Path to the plugin folder.
+        '';
+      };
+
+      mappings = mkOption {
+        type = with types; attrsOf str;
+        description = ''
+          Key mappings to the plugins.
+        '';
+        default = { };
+        example = literalExample ''
+          {
+            c = "fzcd";
+            f = "finder";
+            v = "imgview";
+          };
+        '';
+      };
+    };
+  });
+in {
+  meta.maintainers = with maintainers; [ thiagokokada ];
+
+  options = {
+    programs.nnn = {
+      enable = mkEnableOption "nnn";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.nnn;
+        defaultText = literalExample "pkgs.nnn";
+        example =
+          literalExample "pkgs.nnn.override ({ withNerdIcons = true; });";
+        description = ''
+          Package containing the <command>nnn</command> program.
+        '';
+      };
+
+      finalPackage = mkOption {
+        type = types.package;
+        readOnly = true;
+        visible = false;
+        description = ''
+          Resulting nnn package.
+        '';
+      };
+
+      bookmarks = mkOption {
+        type = with types; attrsOf str;
+        description = ''
+          Directory bookmarks.
+        '';
+        example = literalExample ''
+          {
+            d = "~/Documents";
+            D = "~/Downloads";
+            p = "~/Pictures";
+            v = "~/Videos";
+          };
+        '';
+        default = { };
+      };
+
+      extraPackages = mkOption {
+        type = with types; listOf package;
+        example =
+          literalExample "with pkgs; [ ffmpegthumbnailer mediainfo sxiv ]";
+        description = ''
+          Extra packages available to nnn.
+        '';
+        default = [ ];
+      };
+
+      plugins = mkOption {
+        type = pluginModule;
+        description = ''
+          Manage nnn plugins.
+        '';
+        default = { };
+      };
+    };
+  };
+
+  config = let
+    nnnPackage = cfg.package.overrideAttrs (oldAttrs: {
+      nativeBuildInputs = (oldAttrs.nativeBuildInputs or [ ])
+        ++ [ pkgs.makeWrapper ];
+      postInstall = concatStringsSep "\n" [
+        (oldAttrs.postInstall or "")
+        ''
```

```suggestion
      postInstall = ''
        ${oldAttrs.postInstall}
        
        wrapProgram ....
      ''
```
I think the `concatStringsSep` can be dropped and replaced by a direction string interpolation.